modules/acquia_cms_common/src/EventSubscriber/ConfigEventsSubscriber.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Drupal\acquia_cms_common\EventSubscriber;
+
+use Drupal\acquia_cms_search\Facade\SearchFacade;
+use Drupal\Core\Config\ConfigCrudEvent;
+use Drupal\Core\Config\ConfigEvents;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Class EntityTypeSubscriber.
+ *
+ * @package Drupal\acquia_cms_common\EventSubscriber
+ */
+class ConfigEventsSubscriber implements EventSubscriberInterface {
+
+  /**
+   * {@inheritdoc}
+   *
+   * @return array
+   *   The event names to listen for, and the methods that should be executed.
+   */
+  public static function getSubscribedEvents() {
+    return [
+      ConfigEvents::SAVE => 'configSave',
+    ];
+  }
+
+  /**
+   * React to a config object being saved.
+   *
+   * @param \Drupal\Core\Config\ConfigCrudEvent $event
+   *   Config crud event.
+   */
+  public function configSave(ConfigCrudEvent $event) {
+    $config = $event->getConfig();
+    $module_handler = \Drupal::moduleHandler();
+    // Update views display_options's style type as cohesion_layout and
+    // option's views_template & master_template if acquia_cms_site_studio
+    // module is present else use default one.
+    if ($module_handler->moduleExists('acquia_cms_site_studio')) {
+      switch ($config->getName()) {
+        case 'views.view.event_cards':
+          \Drupal::classResolver(SearchFacade::class)->updateViewDisplayOptionsStyle('event_cards', 'upcoming_events_block', 'view_tpl_event_cards_slider');
```

Comment:
@chandan-singh7929 This (`\Drupal::classResolver(SearchFacade::class`) can go in class constructor.